packages/worker/src/core.ts

```diff
@@ -253,3 +260,65 @@ export class Core {
     delete Core._instance;
   }
 }
+
+export class ReplayCore extends Core {
+  protected static _instance?: Promise<ReplayCore>;
+  protected static instantiator?: 'install' | 'instance';
+  /**
+   * Default options get overridden when Core is installed and are remembered in case Core is
+   * re-instantiated after being shut down
+   */
+  protected static defaultOptions: CoreOptions = {};
+
+  // TODO: How do we de-dupe all this static goop? Maybe have some non-static
+  //   core singleton generic type which these hold a static instance of
+  public static async install(options: CoreOptions): Promise<ReplayCore> {
```

Comment:
You can reuse the original install signature if you use this as a template param:
https://github.com/temporalio/sdk-typescript/blob/afc0d964ecf21bbe5e79b28300e46bd2273dd38e/packages/worker/src/workflow/vm.ts#L77

Reply:
I don't think that works - not without making the entire class generic at least, because there is still only one static `_instance` field and it starts being nonassignable when you do this.

Reply:
Actually I was able to overcome that but it still doesn't work. The static `install` on the base class will use the base class' `_instance` field, not the subclass, so we end up erroneously constructing a real core.